clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -878,6 +878,7 @@ public void assign(Collection<TopicPartition> partitions) {
                 remaining = timeout - elapsed;
             } while (remaining > 0);
 
+            client.quickPoll(false, false);
```

Comment:
Oops, sorry, that was a mistake. I had added that line specifically to fail unsent requests :-)